### DISCUSSION TITLE
Accelerate domestic access without the need for proxy deployment tutorial (国内加速访问，无需代理部署教程)

### DISCUSSION THREAD

**Discussion by yzh990918** [ddiu8081/chatgpt-demo#270] [Ideas]
1. [Fork](https://github.com/ddiu8081/chatgpt-demo/fork) this project，Go to [https://app.netlify.com/start](https://app.netlify.com/start) new Site, select the project you `forked` done, and connect it with your `GitHub` account.

![image](https://cdn.staticaly.com/gh/yzh990918/static@master/20230310/image.3nlt4hgzb16o.webp)

![image](https://cdn.staticaly.com/gh/yzh990918/static@master/20230310/image.5fhfouap270g.webp)

2. Select the branch you want to deploy, then configure environment variables in the project settings.

![image](https://cdn.staticaly.com/gh/yzh990918/static@master/20230311/image.gfs9lx8c854.webp)

3. Select the default build command and output directory, Click the `Deploy Site` button to start deploying the site。

![image](https://cdn.staticaly.com/gh/yzh990918/static@master/20230311/image.4jky9e1wbojk.webp)

4. Prepare a domestic service provider domain name (my side is the domain name of Alibaba Cloud)，Go to [domain name resolution page](https://dns.console.aliyun.com/#/dns/domainList)

![image](https://user-images.githubusercontent.com/46418596/225491762-1d0ffb81-c5e5-4280-bf7a-6333ec81776a.png)

5. Go to the Netlify application details page you just deployed and click the `Domain Settings` button

![image](https://user-images.githubusercontent.com/46418596/225492229-cd7f6186-4d04-4c8c-9d02-3197bb4c38a8.png)

![image](https://user-images.githubusercontent.com/46418596/225495607-db22a182-8a7b-45ef-af90-ebb6bfcf017f.png)

6. Add what you need to parsed domain alias, ex: `ai.xxx.xx`, then copy `CNAME` configuration

![image](https://user-images.githubusercontent.com/46418596/225492672-a495d6f5-29c8-4370-a70f-105873eae934.png)

![image](https://user-images.githubusercontent.com/46418596/225493169-1e39d07b-4cf8-4683-b4fd-38109c6ddd58.png)

7. Go to the resolution settings details page and add a new resolution rule. The host record is the domain name visited, and the record value is the `CNAME` you just copied.

![image](https://user-images.githubusercontent.com/46418596/225496742-be695de7-b515-4b7b-a1da-cb5b69a77461.png)

8. After the addition is complete, go to the SSL certificate page to configure the HTTPS certificate.

![image](https://user-images.githubusercontent.com/46418596/225494068-2c6e6c07-e18e-4ade-b3c7-52f96dcde763.png)

9. Access the domain name you configured, Enjoy yourself 🥰

![image](https://user-images.githubusercontent.com/46418596/225495192-220a0b91-75a8-41f8-9859-a3f9e6c7acec.png)

**Comment by yzh990918** [comment]
Proven effective(亲测有效) 😃

**Reply by Poper789** [reply]
大佬🧍‍♂️

**Reply by daminghh** [reply]
赞

**Reply by yangyj2000** [reply]
为啥我设置了环境变量，但是还是提示我invalid_api_key，求指教

**Reply by Edwardvi** [reply]
> 为啥我设置了环境变量，但是还是提示我invalid_api_key，求指教

我也是，是因为设置字符串写错了吗？你解决了么

**Reply by licc12** [reply]
我也是，不知道怎么解决？楼上解决了吗？

**Reply by licc12** [reply]
我解决了，方案如下：
![image](https://user-images.githubusercontent.com/28993145/230780134-deb7a2a8-4db9-436c-a83a-ef9cd41fea25.png)
上面部署完之后发现key是空的，这时候你需要重新填入你的key即可

**Reply by RankFrank4010** [reply]
@licc12 我的api还是在的，但是任然同样无法问问题

**Reply by xinxvx** [reply]
同样的问题，有解决办法吗

**Comment by Edwardvi** [comment]
你好，这个流程的前半部分换成Vercel的话，是不是可以做到免代理？还是说只适合netlify呢？

**Reply by yzh990918** [reply]
> 你好，这个流程的前半部分换成Vercel的话，是不是可以做到免代理？还是说只适合netlify呢？

Vercel is not applicable, you need to open a "ladder"

**Reply by songhn233** [reply]
> > 你好，这个流程的前半部分换成Vercel的话，是不是可以做到免代理？还是说只适合netlify呢？
> 
> Vercel is not applicable, you need to open a "ladder"

https://vercel.com/docs/concepts/projects/domains/add-a-domain

Vercel also supports custom domains, in case you need to solve DNS pollution issues.

**Reply by erbanku** [reply]
Vercel Hong Kong function region + custom domain can accelerate the domestic access speed and solve the DNS pollution issues. (Tested on my end.)
